packages/sources/coinmetrics-test/src/endpoint/price.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  PriceEndpoint,
+  priceEndpointInputParameters,
+  PriceEndpointParams,
+} from '@chainlink/external-adapter-framework/adapter'
+import { RestTransport } from '@chainlink/external-adapter-framework/transports'
+import { customSettings, DEFAULT_API_ENDPOINT } from '../config'
+import { validateResultNumber } from '../global-utils'
+
+export interface ResponseSchema {
+  data: {
+    asset: string
+    time: string
+    ReferenceRateUSD?: string
+    ReferenceRateEUR?: string
+  }[]
+  error?: {
+    type: string
+    message: string
+  }
+}
+
+type EndpointTypes = {
+  Request: {
+    Params: PriceEndpointParams
+  }
+  Response: {
+    Data: ResponseSchema | undefined
+    Result: number
+  }
+  CustomSettings: typeof customSettings
+  Provider: {
+    RequestBody: never
+    ResponseBody: ResponseSchema
+  }
+}
+
+const restEndpointTransport = new RestTransport<EndpointTypes>({
```

Comment:
These would need to be rewritten to use batches, they will be very inneficient otherwise